### PR TITLE
Hook up fantomas for F# formatting

### PR DESCRIPTION
**Pull request by pbiggar**
Add fantomas to the format script.

Format existing files

Move circleci format checking to its own job (this is clearer for contributors, vs knowing where to hunt down the error message).

Move formatting tools into the docker container (including stashing a static binary of ocamlformat 0.12 to download as part of the build)

### CODE REVIEW

File: fsharp-backend/src/LibBackend/Authorization.fs [hunk 1]
```diff
@@ -152,7 +152,13 @@ let specialCasePermission
...
   (ownerName : OwnerName.T)
   : Option<Permission> =
-  if List.any ((=) (ownerName, username)) specialCases then
+  if List.any
+       ((=)
+         (
+           ownerName,
+           username
+         ))
+       specialCases then
```
Comment (StachuDotNet) [inline fsharp-backend/src/LibBackend/Authorization.fs:161]: I'm not experienced with Fantomas, but this seems much harder to read to me - are you happy with this formatting? (if so, ignore me, I suppose :))

Comment (pbiggar) [reply fsharp-backend/src/LibBackend/Authorization.fs:161]: No, I hate this formatting and I'd love to figure out the switch to not do this.

Comment (StachuDotNet) [reply fsharp-backend/src/LibBackend/Authorization.fs:161]: I love the idea of Fantomas, but was never able to get around this particular formatting awkwardness. I ended up not using Fantomas for my project for this reason. May be worth reaching out to Fantomas repo? I haven't seen a particular setting for this (checked after my above post) :(

Comment (pbiggar) [reply fsharp-backend/src/LibBackend/Authorization.fs:161]: https://github.com/fsprojects/fantomas/issues/1700

**Pull request merged**